app/src/main/java/com/example/countryinfo/CountriesPreviewViewModel.kt

```diff
@@ -0,0 +1,59 @@
+package com.example.countryinfo
+
+import android.util.Log
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.apollographql.apollo.api.Response
+import com.example.GetCountriesQuery
+import com.google.gson.Gson
+import io.reactivex.Observer
+import io.reactivex.disposables.Disposable
+import org.json.JSONObject
+
+class CountriesPreviewViewModel(private val countryApi: ICountriesApi) : ViewModel() {
+    private val TAG = "CountriesPreviewViewMod"
+
+    init {
+        getCountries()
+    }
+
+    private val countriesMutableLiveData = MutableLiveData<CountriesPreviewViewState>()
+    val pollLiveData: LiveData<CountriesPreviewViewState> = countriesMutableLiveData
+
+    private fun getCountries() {
+        val myObservable = countryApi.getCountries()
+        val myObserver = object : Observer<Response<GetCountriesQuery.Data>> {
+            override fun onComplete() {
+                Log.d(TAG, "onComplete: ")
+            }
+
+            override fun onSubscribe(d: Disposable) {
+                Log.d(TAG, "onSubscribe: ")
+            }
+
+            override fun onNext(t: Response<GetCountriesQuery.Data>) {
+                val data = t.data?.country?.map {
+                    country -> CountryPreview(
+                    country?.fragments?.countryPreview?.name?:"",
+                    country?.fragments?.countryPreview?.capital?:"",
+                    country?.fragments?.countryPreview?.flag?.svgFile?:"",
+                    country?.fragments?.countryPreview?.subregion?.region?.name?:""
+                )
+                }
+                countriesMutableLiveData.postValue(
+                    data?.let { CountriesPreviewViewState.Default(it) }
+                )
+            }
+
+            override fun onError(e: Throwable) {
+                Log.d(TAG, "onError: ", e)
+                countriesMutableLiveData.postValue(
+                    CountriesPreviewViewState.Error
+                )
+            }
+        }
+        myObservable.subscribe(myObserver)
```

Comment:
The subscribe method usually returns a reference to `Disposable` which must be `disposed` in the `ViewModel::onCleared()` method (https://developer.android.com/reference/androidx/lifecycle/ViewModel#onCleared()) in order to avoid memory leaks, since observer will keep a reference to ViewModel inside your `onNext` method. 

But when we pass `Observable`  to `subscribe` method it returns `void` which is a pain (you can read more about it here https://medium.com/mobile-app-development-publication/rxjava-2-wheres-disposable-when-subscribe-observer-4ec16049f2e1). So, it's preferable to use overloaded subscribe method which takes just `onNext` and `onError` `Consumer`s. 

http://reactivex.io/RxJava/javadoc/

<img width="1247" alt="Screenshot 2020-10-30 at 11 08 33" src="https://user-images.githubusercontent.com/14247172/97681069-5ea6a080-1aa0-11eb-805b-7af5582c6862.png">

Reply:
Done.